<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Grouping feature groups in components — component-generation • patRoon</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Grouping feature groups in components — component-generation" />
<meta property="og:description" content="Functionality to automatically group related feature groups (e.g. isotopes, adducts and homologues) to assist
and simplify compound annotation." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">patRoon</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.0.0.9999</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Reference
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../reference/index.html">Online</a>
    </li>
    <li>
      <a href="../reference/patRoon.pdf">PDF</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Tutorial
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/tutorial.html">Online HTML page</a>
    </li>
    <li>
      <a href="../articles/tutorial.pdf">PDF</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Handbook
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../handbook_bd/index.html">Book format</a>
    </li>
    <li>
      <a href="../articles/handbook.html">Single page</a>
    </li>
    <li>
      <a href="../handbook_bd/handbook.pdf">PDF</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/rickhelmus/patRoon">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Grouping feature groups in components</h1>
    <small class="dont-index">Source: <a href='https://github.com/rickhelmus/patRoon/blob/master/R/main.R'><code>R/main.R</code></a>, <a href='https://github.com/rickhelmus/patRoon/blob/master/R/components.R'><code>R/components.R</code></a>, <a href='https://github.com/rickhelmus/patRoon/blob/master/R/components-camera.R'><code>R/components-camera.R</code></a>, and 7 more</small>
    <div class="hidden name"><code>component-generation.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Functionality to automatically group related feature groups (<em>e.g.</em> isotopes, adducts and homologues) to assist
and simplify compound annotation.</p>
    </div>

    <pre class="usage"><span class='co'># S4 method for featureGroups</span>
<span class='fu'>generateComponents</span><span class='op'>(</span><span class='va'>fGroups</span>, <span class='va'>algorithm</span>, <span class='va'>...</span><span class='op'>)</span>

<span class='co'># S4 method for featureGroups</span>
<span class='fu'>generateComponentsCAMERA</span><span class='op'>(</span>
  <span class='va'>fGroups</span>,
  ionization <span class='op'>=</span> <span class='cn'>NULL</span>,
  onlyIsotopes <span class='op'>=</span> <span class='cn'>FALSE</span>,
  minSize <span class='op'>=</span> <span class='fl'>2</span>,
  relMinReplicates <span class='op'>=</span> <span class='fl'>0.5</span>,
  extraOpts <span class='op'>=</span> <span class='cn'>NULL</span>
<span class='op'>)</span>

<span class='co'># S4 method for featureGroupsSet</span>
<span class='fu'>generateComponentsCAMERA</span><span class='op'>(</span><span class='va'>fGroups</span>, ionization <span class='op'>=</span> <span class='cn'>NULL</span>, <span class='va'>...</span><span class='op'>)</span>

<span class='co'># S4 method for featureGroups</span>
<span class='fu'>generateComponentsCliqueMS</span><span class='op'>(</span>
  <span class='va'>fGroups</span>,
  ionization <span class='op'>=</span> <span class='cn'>NULL</span>,
  maxCharge <span class='op'>=</span> <span class='fl'>1</span>,
  maxGrade <span class='op'>=</span> <span class='fl'>2</span>,
  ppm <span class='op'>=</span> <span class='fl'>10</span>,
  adductInfo <span class='op'>=</span> <span class='cn'>NULL</span>,
  absMzDev <span class='op'>=</span> <span class='fl'>0.005</span>,
  minSize <span class='op'>=</span> <span class='fl'>2</span>,
  relMinAdductAbundance <span class='op'>=</span> <span class='fl'>0.75</span>,
  adductConflictsUsePref <span class='op'>=</span> <span class='cn'>TRUE</span>,
  NMConflicts <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"preferential"</span>, <span class='st'>"mostAbundant"</span>, <span class='st'>"mostIntense"</span><span class='op'>)</span>,
  prefAdducts <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"[M+H]+"</span>, <span class='st'>"[M-H]-"</span><span class='op'>)</span>,
  extraOptsCli <span class='op'>=</span> <span class='cn'>NULL</span>,
  extraOptsIso <span class='op'>=</span> <span class='cn'>NULL</span>,
  extraOptsAnn <span class='op'>=</span> <span class='cn'>NULL</span>,
  parallel <span class='op'>=</span> <span class='cn'>TRUE</span>
<span class='op'>)</span>

<span class='co'># S4 method for featureGroupsSet</span>
<span class='fu'>generateComponentsCliqueMS</span><span class='op'>(</span><span class='va'>fGroups</span>, ionization <span class='op'>=</span> <span class='cn'>NULL</span>, <span class='va'>...</span><span class='op'>)</span>

<span class='co'># S4 method for featureGroups</span>
<span class='fu'>generateComponentsIntClust</span><span class='op'>(</span>
  <span class='va'>fGroups</span>,
  method <span class='op'>=</span> <span class='st'>"complete"</span>,
  metric <span class='op'>=</span> <span class='st'>"euclidean"</span>,
  normFunc <span class='op'>=</span> <span class='va'>max</span>,
  average <span class='op'>=</span> <span class='cn'>TRUE</span>,
  maxTreeHeight <span class='op'>=</span> <span class='fl'>1</span>,
  deepSplit <span class='op'>=</span> <span class='cn'>TRUE</span>,
  minModuleSize <span class='op'>=</span> <span class='fl'>1</span>
<span class='op'>)</span>

<span class='co'># S4 method for featureGroups</span>
<span class='fu'>generateComponentsNontarget</span><span class='op'>(</span>
  <span class='va'>fGroups</span>,
  ionization <span class='op'>=</span> <span class='cn'>NULL</span>,
  rtRange <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='op'>-</span><span class='fl'>120</span>, <span class='fl'>120</span><span class='op'>)</span>,
  mzRange <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>5</span>, <span class='fl'>120</span><span class='op'>)</span>,
  elements <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"C"</span>, <span class='st'>"H"</span>, <span class='st'>"O"</span><span class='op'>)</span>,
  rtDev <span class='op'>=</span> <span class='fl'>30</span>,
  absMzDev <span class='op'>=</span> <span class='fl'>0.002</span>,
  absMzDevLink <span class='op'>=</span> <span class='va'>absMzDev</span> <span class='op'>*</span> <span class='fl'>2</span>,
  traceHack <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/all.html'>all</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/Version.html'>R.Version</a></span><span class='op'>(</span><span class='op'>)</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"major"</span>, <span class='st'>"minor"</span><span class='op'>)</span><span class='op'>]</span> <span class='op'>&gt;=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>3</span>, <span class='fl'>4</span><span class='op'>)</span><span class='op'>)</span>,
  <span class='va'>...</span>
<span class='op'>)</span>

<span class='co'># S4 method for featureGroupsSet</span>
<span class='fu'>generateComponentsNontarget</span><span class='op'>(</span><span class='va'>fGroups</span>, ionization <span class='op'>=</span> <span class='cn'>NULL</span>, <span class='va'>...</span><span class='op'>)</span>

<span class='co'># S4 method for featureGroups</span>
<span class='fu'>generateComponentsOpenMS</span><span class='op'>(</span>
  <span class='va'>fGroups</span>,
  ionization <span class='op'>=</span> <span class='cn'>NULL</span>,
  chargeMin <span class='op'>=</span> <span class='fl'>1</span>,
  chargeMax <span class='op'>=</span> <span class='fl'>1</span>,
  chargeSpan <span class='op'>=</span> <span class='fl'>3</span>,
  qTry <span class='op'>=</span> <span class='st'>"heuristic"</span>,
  potentialAdducts <span class='op'>=</span> <span class='cn'>NULL</span>,
  minRTOverlap <span class='op'>=</span> <span class='fl'>0.66</span>,
  retWindow <span class='op'>=</span> <span class='fl'>1</span>,
  absMzDev <span class='op'>=</span> <span class='fl'>0.005</span>,
  minSize <span class='op'>=</span> <span class='fl'>2</span>,
  relMinAdductAbundance <span class='op'>=</span> <span class='fl'>0.75</span>,
  adductConflictsUsePref <span class='op'>=</span> <span class='cn'>TRUE</span>,
  NMConflicts <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"preferential"</span>, <span class='st'>"mostAbundant"</span>, <span class='st'>"mostIntense"</span><span class='op'>)</span>,
  prefAdducts <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"[M+H]+"</span>, <span class='st'>"[M-H]-"</span><span class='op'>)</span>,
  extraOpts <span class='op'>=</span> <span class='cn'>NULL</span>
<span class='op'>)</span>

<span class='co'># S4 method for featureGroupsSet</span>
<span class='fu'>generateComponentsOpenMS</span><span class='op'>(</span>
  <span class='va'>fGroups</span>,
  ionization <span class='op'>=</span> <span class='cn'>NULL</span>,
  chargeMin <span class='op'>=</span> <span class='fl'>1</span>,
  chargeMax <span class='op'>=</span> <span class='fl'>1</span>,
  chargeSpan <span class='op'>=</span> <span class='fl'>3</span>,
  qTry <span class='op'>=</span> <span class='st'>"heuristic"</span>,
  potentialAdducts <span class='op'>=</span> <span class='cn'>NULL</span>,
  <span class='va'>...</span>
<span class='op'>)</span>

<span class='fu'>defaultOpenMSAdducts</span><span class='op'>(</span><span class='va'>ionization</span><span class='op'>)</span>

<span class='co'># S4 method for featureGroups</span>
<span class='fu'>generateComponentsRAMClustR</span><span class='op'>(</span>
  <span class='va'>fGroups</span>,
  ionization <span class='op'>=</span> <span class='cn'>NULL</span>,
  st <span class='op'>=</span> <span class='cn'>NULL</span>,
  sr <span class='op'>=</span> <span class='cn'>NULL</span>,
  maxt <span class='op'>=</span> <span class='fl'>12</span>,
  hmax <span class='op'>=</span> <span class='fl'>0.3</span>,
  normalize <span class='op'>=</span> <span class='st'>"TIC"</span>,
  absMzDev <span class='op'>=</span> <span class='fl'>0.002</span>,
  relMzDev <span class='op'>=</span> <span class='fl'>5</span>,
  minSize <span class='op'>=</span> <span class='fl'>2</span>,
  relMinReplicates <span class='op'>=</span> <span class='fl'>0.5</span>,
  RCExperimentVals <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>design <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>platform <span class='op'>=</span> <span class='st'>"LC-MS"</span><span class='op'>)</span>, instrument <span class='op'>=</span>
    <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>ionization <span class='op'>=</span> <span class='va'>ionization</span>, MSlevs <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span>,
  extraOptsRC <span class='op'>=</span> <span class='cn'>NULL</span>,
  extraOptsFM <span class='op'>=</span> <span class='cn'>NULL</span>
<span class='op'>)</span>

<span class='co'># S4 method for featureGroupsSet</span>
<span class='fu'>generateComponentsRAMClustR</span><span class='op'>(</span><span class='va'>fGroups</span>, ionization <span class='op'>=</span> <span class='cn'>NULL</span>, <span class='va'>...</span><span class='op'>)</span>

<span class='co'># S4 method for featureGroups</span>
<span class='fu'>generateComponentsSpecClust</span><span class='op'>(</span>
  <span class='va'>fGroups</span>,
  <span class='va'>MSPeakLists</span>,
  method <span class='op'>=</span> <span class='st'>"complete"</span>,
  specSimParams <span class='op'>=</span> <span class='fu'><a href='specSimParams.html'>getDefSpecSimParams</a></span><span class='op'>(</span><span class='op'>)</span>,
  maxTreeHeight <span class='op'>=</span> <span class='fl'>1</span>,
  deepSplit <span class='op'>=</span> <span class='cn'>TRUE</span>,
  minModuleSize <span class='op'>=</span> <span class='fl'>1</span>
<span class='op'>)</span>

<span class='co'># S4 method for featureGroups</span>
<span class='fu'>generateComponentsTPs</span><span class='op'>(</span>
  <span class='va'>fGroups</span>,
  fGroupsTPs <span class='op'>=</span> <span class='va'>fGroups</span>,
  ignoreParents <span class='op'>=</span> <span class='cn'>FALSE</span>,
  TPs <span class='op'>=</span> <span class='cn'>NULL</span>,
  MSPeakLists <span class='op'>=</span> <span class='cn'>NULL</span>,
  formulas <span class='op'>=</span> <span class='cn'>NULL</span>,
  compounds <span class='op'>=</span> <span class='cn'>NULL</span>,
  minRTDiff <span class='op'>=</span> <span class='fl'>20</span>,
  specSimParams <span class='op'>=</span> <span class='fu'><a href='specSimParams.html'>getDefSpecSimParams</a></span><span class='op'>(</span><span class='op'>)</span>
<span class='op'>)</span>

<span class='co'># S4 method for featureGroupsSet</span>
<span class='fu'>generateComponentsTPs</span><span class='op'>(</span>
  <span class='va'>fGroups</span>,
  fGroupsTPs <span class='op'>=</span> <span class='va'>fGroups</span>,
  ignoreParents <span class='op'>=</span> <span class='cn'>FALSE</span>,
  TPs <span class='op'>=</span> <span class='cn'>NULL</span>,
  MSPeakLists <span class='op'>=</span> <span class='cn'>NULL</span>,
  formulas <span class='op'>=</span> <span class='cn'>NULL</span>,
  compounds <span class='op'>=</span> <span class='cn'>NULL</span>,
  minRTDiff <span class='op'>=</span> <span class='fl'>20</span>,
  specSimParams <span class='op'>=</span> <span class='fu'><a href='specSimParams.html'>getDefSpecSimParams</a></span><span class='op'>(</span><span class='op'>)</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>fGroups</th>
      <td><p><code><a href='featureGroups-class.html'>featureGroups</a></code> object for which components should be generated.</p>
<p>For <code>generateComponentsTPs</code>: also see the <code>fGroupsTPs</code> argument.</p></td>
    </tr>
    <tr>
      <th>algorithm</th>
      <td><p>A character string describing the algorithm that should be
used: <code>"ramclustr"</code>, <code>"camera"</code>, <code>"nontarget"</code>, <code>"intclust"</code>, <code>"openms"</code>, <code>"cliquems"</code>, <code>"specclust"</code>, <code>"tp"</code></p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Any parameters to be passed to the selected component generation algorithm.</p>
<p>For sets workflows: further arguments directly passed to the non-sets method.</p></td>
    </tr>
    <tr>
      <th>ionization</th>
      <td><p>Which ionization polarity was used to generate the data: should be <code>"positive"</code> or
  <code>"negative"</code>. If the <code>featureGroups</code> object has adduct annotations, and <code>ionization=NULL</code>, the
  ionization will be detected automatically.</p>
<p>(<strong>sets workflow</strong>) This parameter is not supported for sets workflows, as the ionization will always be detected
  automatically.</p></td>
    </tr>
    <tr>
      <th>onlyIsotopes</th>
      <td><p>Logical value. If <code>TRUE</code> only isotopes are
considered when generating components (faster). Corresponds to <code>quick</code>
argument of <code><a href='https://rdrr.io/pkg/CAMERA/man/annotate-methods.html'>CAMERA::annotate</a></code>.</p></td>
    </tr>
    <tr>
      <th>minSize</th>
      <td><p>The minimum size of a component. Smaller components than this size will be removed. For
<code>RAMClustR</code>: sets the <code>minModuleSize</code> argument to <code><a href='https://rdrr.io/pkg/RAMClustR/man/ramclustR.html'>ramclustR</a></code>. See note below.</p></td>
    </tr>
    <tr>
      <th>relMinReplicates</th>
      <td><p>Feature groups within a component are only kept when they contain data for at least this
(relative) amount of replicate analyses. For instance, <samp>0.5</samp> means that at least half of the replicates should
contain data for a particular feature group in a component. In this calculation replicates that are fully absent
within a component are not taken in to account. See note below.</p></td>
    </tr>
    <tr>
      <th>extraOpts</th>
      <td><p>Named character vector with extra arguments directly passed to
<code><a href='https://rdrr.io/pkg/CAMERA/man/annotate-methods.html'>CAMERA::annotate</a></code> (<code>generateComponentsCAMERA</code>) or as extra command line
parameters to <code class='command'>MetaboliteAdductDecharger</code> (<code>generateComponentsCAMERA</code>). Set to <code>NULL</code> to ignore.</p></td>
    </tr>
    <tr>
      <th>maxCharge, maxGrade, ppm</th>
      <td><p>Arguments passed to <code><a href='https://rdrr.io/pkg/cliqueMS/man/getIsotopes.html'>cliqueMS::getIsotopes</a></code> and/or
<code><a href='https://rdrr.io/pkg/cliqueMS/man/getAnnotation.html'>cliqueMS::getAnnotation</a></code>.</p></td>
    </tr>
    <tr>
      <th>adductInfo</th>
      <td><p>Sets the <code>adinfo</code> argument to <code><a href='https://rdrr.io/pkg/cliqueMS/man/getAnnotation.html'>cliqueMS::getAnnotation</a></code>.
If <code>NULL</code> then the default adduct information from <span class="pkg">cliqueMS</span> is used (<em>i.e.</em> the
<code>positive.adinfo</code>/<code>negative.adinfo</code> package datasets).</p></td>
    </tr>
    <tr>
      <th>relMinAdductAbundance</th>
      <td><p>The minimum relative abundance (<samp>0-1</samp>) that an adduct should be assigned to
features within the same feature group. See the <code>Feature components</code> section for more details.</p></td>
    </tr>
    <tr>
      <th>adductConflictsUsePref</th>
      <td><p>If set to <code>TRUE</code>, and not all adduct assigments to the features within a feature
group are equal and at least one of those adducts is a preferential adduct (<code>prefAdducts</code> argument), then only
the features with (the lowest ranked) preferential adduct are considered. In all other cases or when
<code>adductConflictsUsePref=FALSE</code> only features with the most frequently assigned adduct is considered. See the
<code>Feature components</code> section for more details.</p></td>
    </tr>
    <tr>
      <th>NMConflicts</th>
      <td><p>The strategies to employ when not all neutral masses within a component are equal. Valid options
are: <code>"preferential"</code>, <code>"mostAbundant"</code> and <code>"mostIntense"</code>. Multiple strategies are possible, and
will be executed in the given order until one succeeds. See the <code>Feature components</code> section for more details.</p></td>
    </tr>
    <tr>
      <th>prefAdducts</th>
      <td><p>A <code>character</code> vector with one or more <em>preferential adducts</em>. See the <code>Feature
components</code> section for more details.</p></td>
    </tr>
    <tr>
      <th>extraOptsCli, extraOptsIso, extraOptsAnn</th>
      <td><p>Named <code>list</code> with further arguments to be passed to
<code><a href='https://rdrr.io/pkg/cliqueMS/man/getCliques.html'>cliqueMS::getCliques</a></code>, <code><a href='https://rdrr.io/pkg/cliqueMS/man/getIsotopes.html'>cliqueMS::getIsotopes</a></code>
and <code><a href='https://rdrr.io/pkg/cliqueMS/man/getAnnotation.html'>cliqueMS::getAnnotation</a></code>, respectively. Set to <code>NULL</code> to ignore.</p></td>
    </tr>
    <tr>
      <th>parallel</th>
      <td><p>If set to <code>TRUE</code> then code is executed in parallel through the <a href='https://CRAN.R-project.org/package=futures'><span class="pkg">futures</span></a> package. Please
see the parallelization section in the handbook for more details.</p></td>
    </tr>
    <tr>
      <th>method</th>
      <td><p>Clustering method that should be applied (passed to
<code><a href='https://rdrr.io/pkg/fastcluster/man/hclust.html'>fastcluster::hclust</a></code>).</p></td>
    </tr>
    <tr>
      <th>metric</th>
      <td><p>Distance metric used to calculate the distance matrix (passed to <code>daisy</code>).</p></td>
    </tr>
    <tr>
      <th>normFunc, average</th>
      <td><p>Passed to <code><a href='featureGroups-class.html'>as.data.table</a></code> to perform
normalization and averaging of data.</p></td>
    </tr>
    <tr>
      <th>maxTreeHeight, deepSplit, minModuleSize</th>
      <td><p>Arguments used by
<code>cutreeDynamicTree</code>.</p></td>
    </tr>
    <tr>
      <th>rtRange</th>
      <td><p>A numeric vector containing the minimum and maximum retention
time (in seconds) between homologues. Series are always considered from low
to high <em>m/z</em>, thus, a negative minimum retention time allows
detection of homologous series with increasing <em>m/z</em> and decreasing
retention times. These values set the <code>minrt</code> and <code>maxrt</code>
arguments of <code>homol.search</code>.</p></td>
    </tr>
    <tr>
      <th>mzRange</th>
      <td><p>A numeric vector specifying the minimum and maximum <em>m/z</em>
increment of a homologous series. Sets the <code>minmz</code> and <code>maxmz</code>
arguments of <code>homol.search</code>.</p></td>
    </tr>
    <tr>
      <th>elements</th>
      <td><p>A character vector with elements to be considered for
detection of repeating units. Sets the <code>elements</code> argument of
<code>homol.search</code> function.</p></td>
    </tr>
    <tr>
      <th>rtDev, absMzDev, relMzDev</th>
      <td><p>Maximum deviation for retention time or absolute/relative <em>m/z</em>.</p>
<p>For <code>generateComponentsRAMClustR</code>: Sets the <code>mzabs.error</code> and <code>ppm.error</code> arguments to
  <code><a href='https://rdrr.io/pkg/RAMClustR/man/do.findmain.html'>do.findmain</a></code>.</p>
<p>For <code>generateComponentsNontarget</code>: Sets the <code>rttol</code> and <code>mztol</code> arguments of
  <code>homol.search</code>.</p>
<p>For <code>generateComponentsOpenMS</code>: Sets the <code class='command'>algorithm:MetaboliteFeatureDeconvolution:mass_max_diff</code>
  option.</p></td>
    </tr>
    <tr>
      <th>absMzDevLink</th>
      <td><p>Maximum absolute <em>m/z</em> deviation when linking
series. This should usually be a bit higher than <code>absMzDev</code> to ensure
proper linkage.</p></td>
    </tr>
    <tr>
      <th>traceHack</th>
      <td><p>Currently <code>homol.search</code> does not work with <span style="R">R</span>
<samp>&gt;3.3.3</samp>. This flag, which is enabled by default on these R versions,
implements a (messy) workaround
(<a href='https://github.com/blosloos/nontarget/issues/6'>more details here</a>).</p></td>
    </tr>
    <tr>
      <th>chargeMin, chargeMax</th>
      <td><p>The minimum/maximum charge to consider. Corresponds to the
<code class='command'>algorithm:MetaboliteFeatureDeconvolution:charge_min</code>/<code class='command'>algorithm:MetaboliteFeatureDeconvolution:charge_min</code>
 options.</p></td>
    </tr>
    <tr>
      <th>chargeSpan</th>
      <td><p>The maximum charge span for a single analyte. Corresponds to
<code class='command'>algorithm:MetaboliteFeatureDeconvolution:charge_span_max</code>.</p></td>
    </tr>
    <tr>
      <th>qTry</th>
      <td><p>Sets how charges are determined. Corresponds to <code class='command'>algorithm:MetaboliteFeatureDeconvolution:q_try</code>.
Valid options are <code>"heuristic"</code> and <code>"all"</code> (the <code>"feature"</code> option from <code class='command'>OpenMS</code> is
currently not supported).</p></td>
    </tr>
    <tr>
      <th>potentialAdducts</th>
      <td><p>The adducts to consider. Should be a <code>numeric</code> vector with probabilities for each
  adduct, <em>e.g.</em> <code>potentialAdducts=c("[M+H]+" = 0.8, "[M+Na]+" = 0.2)</code>. Note that the sum of probabilities
  should always be <samp>1</samp>. Furthermore, note that additions of multiple adducts should be controlled by the
  <code>chargeMin</code>/<code>chargeMax</code> arguments (and <em>not</em> with <code>potentialAdducts</code>), <em>e.g.</em> if
  <code>chargeMax=2</code> then both <code>[M+H]+</code> and <code>[2M+H]2+</code> may be considered. Please see the
  <code class='command'>algorithm:MetaboliteFeatureDeconvolution:potential_adducts</code> option of
  <a href='https://abibuilder.informatik.uni-tuebingen.de/archive/openms/Documentation/release/latest/html/UTILS_MetaboliteAdductDecharger.html'>MetaboliteAdductDecharger</a>
   for more details. If <code>NULL</code> then the a default is chosen with <code>defaultOpenMSAdducts</code> (which is
  <em>not</em> the same as <code class='command'>OpenMS</code>).</p>
<p>(<strong>sets workflow</strong>) Should be a <code>list</code> where each entry specifies the potential adducts for a set. Should either be named
  with the sets names or follow the same order as <code><a href='generics.html'>sets(fGroups)</a></code>. Example:
  <code>potentialAdducts=list(positive=c("[M+H]+" = 0.8, "[M+Na]+" = 0.2), negative=c("[M-H]-" = 0.8, "[M-H2O-H]-" = 0.2))</code></p></td>
    </tr>
    <tr>
      <th>minRTOverlap, retWindow</th>
      <td><p>Sets feature retention tolerances when grouping features. Sets the
<code class='command'>"algorithm:MetaboliteFeatureDeconvolution:retention_max_diff"</code> and
<code class='command'>algorithm:MetaboliteFeatureDeconvolution:min_rt_overlap</code> options.</p></td>
    </tr>
    <tr>
      <th>st, sr, maxt, hmax, normalize</th>
      <td><p>Arguments to tune the behaviour of feature
group clustering. See their documentation from <code><a href='https://rdrr.io/pkg/RAMClustR/man/ramclustR.html'>ramclustR</a></code>.
When <code>st</code> is <code>NULL</code> it will be automatically calculated as the
half of the median for all chromatographic peak widths.</p></td>
    </tr>
    <tr>
      <th>RCExperimentVals</th>
      <td><p>A named <code>list</code> containing two more <code>list</code>s:
<code>design</code> and <code>instrument</code>. These are used to construct the
<code>ExpDes</code> argument passed to <code><a href='https://rdrr.io/pkg/RAMClustR/man/ramclustR.html'>ramclustR</a></code>.</p></td>
    </tr>
    <tr>
      <th>extraOptsRC, extraOptsFM</th>
      <td><p>Named <code>list</code> with further arguments to be
passed to <code><a href='https://rdrr.io/pkg/RAMClustR/man/ramclustR.html'>ramclustR</a></code> and <code><a href='https://rdrr.io/pkg/RAMClustR/man/do.findmain.html'>do.findmain</a></code>.
Set to <code>NULL</code> to ignore.</p></td>
    </tr>
    <tr>
      <th>MSPeakLists</th>
      <td><p>The <code><a href='MSPeakLists-class.html'>MSPeakLists</a></code> object for the given feature groups that should be used for MS
spectral similarity calculations. For <code>generateComponentsTPs</code> it can be <code>NULL</code>, in which case no
calculations are performed.</p></td>
    </tr>
    <tr>
      <th>specSimParams</th>
      <td><p>A named <code>list</code> with parameters that influence the calculation of MS spectra similarities.
See the <a href='specSimParams.html'>spectral similarity parameters</a> documentation for more details.</p></td>
    </tr>
    <tr>
      <th>fGroupsTPs</th>
      <td><p>A <code><a href='featureGroups-class.html'>featureGroups</a></code> object containing the feature groups that are expected to be
transformation products. If a distinction between parents and TPs is not yet known, <code>fGroupsTPs</code> should equal
the <code>fGroups</code> argument. Otherwise, <code>fGroups</code> should only contain the parent feature groups, and both
<code>fGroups</code> and <code>fGroupsTPs</code> <em>must</em> be a subset of the same <code><a href='featureGroups-class.html'>featureGroups</a></code> object.</p></td>
    </tr>
    <tr>
      <th>ignoreParents</th>
      <td><p>If <code>TRUE</code> then feature groups present in both <code>fGroups</code> and <code>fGroupsTPs</code> are not
considered as TPs.</p></td>
    </tr>
    <tr>
      <th>TPs</th>
      <td><p>A <code><a href='transformationProducts-class.html'>transformationProducts</a></code> object. Set to <code>NULL</code> to perform linking without this data.</p></td>
    </tr>
    <tr>
      <th>formulas, compounds</th>
      <td><p>A <code><a href='formulas-class.html'>formulas</a></code>/<code><a href='compounds-class.html'>compounds</a></code> object to calculate annotation
similarities between parents and TPs. If <code>NULL</code> then this data is not calculated. For more details see the
<code>Linking parents and transformation products</code> section below.</p></td>
    </tr>
    <tr>
      <th>minRTDiff</th>
      <td><p>Minimum retention time (in seconds) difference between the parent and a TP to determine whether a TP
elutes prior/after the parent (to calculate <code>retDir</code> values, see Details in <a href='componentsTPs-class.html'>componentsTPs</a>))</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A <code><a href='length,transformationProducts-method.html'>components</a></code> (derived) object containing all generated components.</p>
<p>The algorithms based on 'feature components' return a <code><a href='length,transformationProducts-method.html'>componentsFeatures</a></code> derived object.</p>
<p><code>generateComponentsIntClust</code> and <code>generateComponentsSpecClust</code> return objects derived from
  <code><a href='componentsSpecClust-class.html'>componentsSpecClust</a></code>.</p>
<p><code>generateComponentsTPs</code> returns objects derived from <code><a href='componentsTPs-class.html'>componentsTPs</a></code>.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Several algorithms are provided to group feature groups that are related in some (chemical) way to each other. These
components generally include adducts, isotopes, in-source fragments and homologues. The linking of this data is
generally useful to provide more information for compound annotation and reduce the data size and thus complexity.</p>
<p><code>generateComponents</code> is a generic function that will generate components using one of the supported algorithms. The actual
  functionality is provided by algorithm specific functions such as <code>generateComponentsRAMClustR</code> and <code>generateComponentsNontarget</code>. While these
  functions may be called directly, <code>generateComponents</code> provides a generic interface and is therefore usually preferred.</p>
<p><code>generateComponentsCAMERA</code> provides an interface to
  <a href='https://bioconductor.org/packages/release/bioc/html/CAMERA.html'>CAMERA</a>
   which is used to generate components from known adducts, isotopes and
  in-source fragments. The specified <code>featureGroups</code> object is
  automatically converted to an <code>xcmsSet</code> object using
  <code><a href='getXCMSSet.html'>getXCMSSet</a></code>.</p>
<p><code>generateComponentsCliqueMS</code> uses <a href='https://github.com/osenan/cliqueMS'>cliqueMS</a> to generate
  components using the <code><a href='https://rdrr.io/pkg/cliqueMS/man/getCliques.html'>cliqueMS::getCliques</a></code> function. The grouping of features in
  each component ('clique') is based on high similarity of chromatographic elution profiles. All features in each
  component are then annotated with the <code><a href='https://rdrr.io/pkg/cliqueMS/man/getIsotopes.html'>cliqueMS::getIsotopes</a></code> and
  <code><a href='https://rdrr.io/pkg/cliqueMS/man/getAnnotation.html'>cliqueMS::getAnnotation</a></code> functions.</p>
<p><code>generateComponentsIntClust</code> generates components based on intensity profiles of feature groups.
  Hierarchical clustering is performed on normalized (and optionally replicate averaged) intensity data and the
  resulting dendrogram is automatically cut with <code>cutreeDynamicTree</code>. The distance matrix is calculated
  with <code>daisy</code> and clustering is performed with <code><a href='https://rdrr.io/pkg/fastcluster/man/hclust.html'>fastcluster::hclust</a></code>.
  The clustering of the resulting components can be further visualized and modified using the methods defined for
  <code><a href='componentsIntClust-class.html'>componentsIntClust</a></code>.</p>
<p><code>generateComponentsNontarget</code> uses
  <a href='https://cran.r-project.org/web/packages/nontarget/index.html'>the
  nontarget R package</a> to generate components by unsupervised detection of
  homologous series. In the first step the <code>homol.search</code>
  function is used to detect all homologues within each replicate group
  (analyses within each replicate group are averaged prior to detection).
  Then, homologous series across replicate groups are merged in case of full
  overlap or when merging of partial overlapping series causes no conflicts.</p>
<p><code>generateComponentsOpenMS</code> uses the
  <a href='https://abibuilder.informatik.uni-tuebingen.de/archive/openms/Documentation/release/latest/html/UTILS_MetaboliteAdductDecharger.html'>MetaboliteAdductDecharger</a>
   utility (see <a href='http://www.openms.de'>http://www.openms.de</a>) to generate components. Features that show highly similar chromatographic
  elution profiles are grouped, and subsequently annotated with their adducts.</p>
<p><code>defaultOpenMSAdducts</code> returns the default adducts and their probabilities when the OpenMS algorithm is
  used for componentization. See the <code>potentialAdducts</code> argument for more details.</p>
<p><code>generateComponentsRAMClustR</code> uses
  <a href='https://github.com/cbroeckl/RAMClustR'>RAMClustR</a> to generate
  components from feature groups which follow similar chromatographic
  retention profiles, but are not necessarily restricted to known rules
  (<em>e.g.</em> adducts or isotopes). This method uses the
  <code><a href='https://rdrr.io/pkg/RAMClustR/man/ramclustR.html'>ramclustR</a></code> functions for generating the components, whereas
  <code><a href='https://rdrr.io/pkg/RAMClustR/man/do.findmain.html'>do.findmain</a></code> is used for annotation.</p>
<p><code>generateComponentsSpecClust</code> generates components based on MS/MS similarity between feature groups.
  The similarities are converted to a distance matrix and used as input for hierarchical clustering, and the
  resulting dendrogram is automatically cut with <code>cutreeDynamicTree</code>. The clustering is performed with
  <code><a href='https://rdrr.io/pkg/fastcluster/man/hclust.html'>fastcluster::hclust</a></code>.</p>
<p><code>generateComponentsTPs</code> generates components by linking feature groups of transformation products and
  their parents. Moreover, this method typically employs data from <a href='TP-generation.html'>generated transformation
  products</a> to find parents and their TPs. However, this data is not necessary, and components can also be made based
  on MS/MS similarity and/or other annotation similarities between the parent and its TPs. For more details see the
  <code>Linking parents and transformation products</code> section below.</p>
    <h2 class="hasAnchor" id="note"><a class="anchor" href="#note"></a>Note</h2>

    <p>For <code>generateComponentsCAMERA</code> and <code>generateComponentsRAMClustR</code>: the <code>minSize</code> and
  <code>relMinReplicates</code> arguments provide additional filtering functionality not provided by <span class="pkg">CAMERA</span> or
  <span class="pkg">RAMClustR</span> (except <code>minSize</code>). Note that these filters are enabled by default, hence, final results may
  be different than what CAMERA/RAMClustR normally would return.</p>
<p>The <code>shift</code> parameter of <code>specSimParams</code> is ignored by <code>generateComponentsTPs</code>, since it always
  calculates similarities with all supported options.</p>
    <h2 class="hasAnchor" id="feature-components"><a class="anchor" href="#feature-components"></a>Feature components</h2>

    <p>Components resulting from <code>generateComponentsCliqueMS</code> and
  <code>generateComponentsOpenMS</code> are based on so called <em>feature components</em>. Unlike other algorithms,
  components are first made on a feature level (per analysis), instead of for complete feature groups. In the final
  step the feature components are converted to 'regular' components by employing a consensus approach with the
  following steps:</p>
<p></p><ol>
<li><p>If an adduct assigned to a feature only occurs as a minority compared to other adduct assigments within the
  same feature group, it is considered as an outlier and removed accordingly (controlled by the
  <code>relMinAdductAbundance</code> argument).</p></li>
<li><p>For features within a feature group, only keep their adduct assignment if it occurs as the most frequent or
  is preferential (controlled by <code>adductConflictsUsePref</code> and <code>prefAdducts</code> arguments).</p></li>
<li><p>Components are made by combining the feature groups for which at least one of their features are jointly
  present in the same feature component.</p></li>
<li><p>Conflicts of neutral mass assignments within a component (<em>i.e.</em> not all are the same) are dealt with.
  Firstly, all feature groups with an unknown neutral mass are split in another component. Then, if conflicts still
  occur, the feature groups with similar neutral mass (determined by <code>absMzDev</code> argument) are grouped. Depending
  on the <code>NMConflicts</code> argument, the group with one or more preferential adduct(s) or that is the largest or
  most intense is selected, whereas others are removed from the component. In case multiple groups contain
  preferential adducts, and <samp>&gt;1</samp> preferential adducts are available, the group with the adduct that matches
  first in <code>prefAdducts</code> 'wins'. In case of ties, one of the next strategies in <code>NMConflicts</code> is tried.</p></li>
<li><p>If a feature group occurs in multiple components it will be removed completely.</p></li>
<li><p>the <code>minSize</code> filter is applied.</p></li>
</ol>

    <h2 class="hasAnchor" id="sets-workflows"><a class="anchor" href="#sets-workflows"></a>Sets workflows</h2>

    <p>In a <a href='sets-workflow.html'>sets workflow</a> the componentization data is generated differently
  depending on the used algorithm:</p><ul>
<li><p>The componentization algorithms that support annotation of adducts/isotopes, <em>e.g.</em> <span class="pkg">RAMClustR</span>,
  <span class="pkg">CAMERA</span> and <code class='command'>OpenMS</code>, first perform componentization for each set independently. This is necessary,
  because <em>e.g.</em> adducts are specific to the MS ionization mode that was used. The resulting components are then
  all combined in a <code><a href='length,transformationProducts-method.html'>componentsSet</a></code> object. Note that the components themselves are never merged. The
  components are renamed to include the set name for which they were generated (<em>e.g.</em> <code>"CMP1"</code> becomes
  <code>"CMP1-positive"</code>).</p></li>
<li><p>For componentization with <span class="pkg">nontarget</span> a similar strategy is applied as above. However, the output is
  stored in a <code><a href='componentsNT-class.html'>componentsNTSet</a></code> object, which supports additional methods such as <code>plotGraph</code>.</p></li>
<li><p>For the componentization based on hierarchical clustering, <em>i.e.</em> the <code>"intclust"</code> and
  <code>"specclust"</code> algorithms, the output format is not different than for a non-sets workflow. However, for
  intensity based clustering, normalization of feature intensities occurs per set. For MS/MS based clustering,
  spectral similarities from each sets are combined as is described for the
  <code><a href='MSPeakLists-class.html'>spectrumSimilarity</a></code> method for sets workflows.</p></li>
<li><p>For componentization of transformation products, the output class is the same (<code><a href='componentsTPs-class.html'>componentsTPs</a></code>).
  However, the component tables are amended with extra information such as overall/specific set spectrum
  similarities. As sets data is mixed, transformation products are able to be linked with a parent, even if they were
  not measured in the same set.</p></li>
</ul>

    <h2 class="hasAnchor" id="parallelization"><a class="anchor" href="#parallelization"></a>Parallelization</h2>

    <p><code>generateComponentsOpenMS</code> uses multiprocessing to parallelize
  computations. Please see the parallelization section in the handbook for
  more details and <a href='patRoon-package.html'>patRoon options</a> for configuration
  options.</p>
<p>Note that for caching purposes, the analyses files must always exist on the local host
  computer, even if it is not participating in computations.</p>
    <h2 class="hasAnchor" id="linking-parents-and-transformation-products"><a class="anchor" href="#linking-parents-and-transformation-products"></a>Linking parents and transformation products</h2>

    <p>With <code>generateComponentsTPs</code>, each component consists of
  feature groups that are considered to be transformation products for one parent (the parent that 'belongs' to the
  component can be retrieved with the <code><a href='length,transformationProducts-method.html'>componentInfo</a></code> method). The parent feature groups are taken from
  the <code>fGroups</code> parameter, while the feature groups for TPs are taken from <code>fGroupsTPs</code>. If a feature group
  occurs in both variables, it may therefore be considered as both a parent or TP.</p>
<p>If transformation product data is given, <em>i.e.</em> the <code>TPs</code> argument is set, then a suspect screening of
  the TPs must be performed in advance (see <code><a href='suspect-screening.html'>screenSuspects</a></code> and <code><a href='generics.html'>convertToSuspects</a></code> to
  create the suspect list). Furthermore, if TPs were generated with <code><a href='TP-generation.html'>generateTPsBioTransformer</a></code> or
  <code><a href='TP-generation.html'>generateTPsLibrary</a></code> then the suspect screening must also include the parents (<em>e.g.</em> by setting
  <code>includeParents=TRUE</code> when calling <code>convertToSuspects</code> or by amending results by setting
  <code>amend=TRUE</code> to <code>screenSuspects</code>). The suspect screening is necessary for the componentization algorithm
  to map the feature groups of the parent or TP. If the the suspect screening yields multiple TP hits, all will be
  reported. Similarly, if the suspect screening contains multiple hits for a parent, a component is made for each of
  the parent hits.</p>
<p>In case no transformation product data is provided (<code>TPs=NULL</code>), the componentization algorithm simply assumes
  that each feature group from <code>fGroupsTPs</code> is a potential TP for every parent feature group in <code>fGroups</code>.
  For this reason, it is highly recommended to specify which feature groups are parents/TPs (see the
  <code>fGroupsTPs</code> argument description above) and <em>crucial</em> that the data is post-processed, for instance by
  only retaining TPs that have high annotation similarity with their parents (see the
  <code><a href='componentsTPs-class.html'>filter</a></code> method for <code><a href='componentsTPs-class.html'>componentsTPs</a></code>).</p>
<p>An typical way to distinguish which feature groups are parents or TPs from two different (groups of) samples is by
  calculating Fold Changes (see the <code><a href='featureGroups-class.html'>as.data.table</a></code> method for
  feature groups and <code><a href='generics.html'>plotVolcano</a></code>). Of course, other statistical techniques from <span style="R">R</span> are also suitable.</p>
<p>During componentization, several characteristics are calculated which may be useful for post-processing:</p><ul>
<li><p><code>specSimilarity</code>: the MS/MS spectral similarity between the feature groups of the TP and its parent
  (<samp>0-1</samp>).</p></li>
<li><p><code>specSimilarityPrec</code>,<code>specSimilarityBoth</code>: as <code>specSimilarity</code>, but calculated with binned
  data using the <code>"precursor"</code> and <code>"both"</code> method, respectively (see <a href='specSimParams.html'>MS spectral
  similarity parameters</a> for more details).</p></li>
<li><p><code>fragmentMatches</code> The number of MS/MS fragment formula annotations that overlap between the TP and
  parent. If both the <code>formulas</code> and <code>compounds</code> arguments are specified then the annotation data is pooled
  prior to calculation. Note that only unique matches are counted. Furthermore, note that annotations from <em>all</em>
  candidates are considered, even if the formula/structure of the parent/TP is known. Hence, <code>fragmentMatches</code>
  is mainly useful when little or no chemical information is known on the parents/TPs, <em>i.e.</em>, when
  <code>TPs=NULL</code> or originates from <code><a href='TP-generation.html'>generateTPsLogic</a></code>. Since annotations for all candidates are used,
  it is highly recommended that the annotation objects are first processed with the <code><a href='generics.html'>filter</a></code> method, for
  instance, to select only the top ranked candidates.</p></li>
<li><p><code>neutralLossMatches</code> As <code>fragmentMatches</code>, but counting overlapping neutral loss formulae.</p></li>
<li><p><code>retDir</code> The retention time direction of the TP relative to its parent. See Details in
  <a href='componentsTPs-class.html'>componentsTPs</a>. If TP data was specified, the expected direction is stored in <code>TP_retDir</code>.</p></li>
<li><p><code>retDiff</code>,<code>mzDiff</code>,<code>formulaDiff</code> The retention time, <em>m/z</em> and formula difference between
  the parent and TP (latter only available if data TP formula is available).</p></li>
</ul>

    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Schollee JE, Bourgin M, von Gunten U, McArdell CS, Hollender J (2018).
&#8220;Non-target screening to trace ozonation transformation products in a wastewater treatment train including different post-treatments.&#8221;
<em>Water Research</em>, <b>142</b>, 267--278.
doi: <a href='https://doi.org/10.1016/j.watres.2018.05.045'>10.1016/j.watres.2018.05.045</a>
.
 <br /><br /> Daniel Müllner (2013). fastcluster: Fast Hierarchical, Agglomerative Clustering Routines for R and Python. Journal of Statistical Software, 53(9), 1-18. URL https://www.jstatsoft.org/v53/i09/.</p>
<p>Kuhl, C., Tautenhahn, R., Boettcher, C., Larson, T. R. and Neumann, S. CAMERA: an integrated strategy for compound spectra extraction and annotation of liquid chromatography/mass spectrometry data sets. Analytical Chemistry, 84:283-289 (2012)</p>
<p>Senan O, Aguilar-Mogas A, Navarro M, Capellades J, Noon L, Burks D, Yanes O, Guimera R, Sales-Pardo M (2019).
&#8220;CliqueMS: a computational tool for annotating in-source metabolite ions from LC-MS untargeted metabolomics data based on a coelution similarity network.&#8221;
<em>Bioinformatics</em>, <b>35</b>(20), 4089--4097.
doi: <a href='https://doi.org/10.1093/bioinformatics/btz207'>10.1093/bioinformatics/btz207</a>
.</p>
<p>Martin Loos (2016). nontarget: Detecting Isotope, Adduct and Homologue Relations in LC-MS Data. R package version 1.9.  <br /><br /> Loos, M., Gerber, C., Corona, F., Hollender, J., Singer, H. (2015). 
			Accelerated isotope fine structure calculation using pruned transition trees, 
			Analytical Chemistry 87(11), 5738-5744.</p>
<p>Bielow C, Ruzek S, Huber CG, Reinert K (2010).
&#8220;Optimal Decharging and Clustering of Charge Ladders Generated in ESI-MS.&#8221;
<em>Journal of Proteome Research</em>, <b>9</b>(5), 2688--2695.
doi: <a href='https://doi.org/10.1021/pr100177k'>10.1021/pr100177k</a>
.</p>
<p>Broeckling, Heuberger CD;, Prince AL;, Ingelsson JA;, Prenni E;, E. J (2013).
&#8220;Assigning precursor-product ion relationships in indiscriminant MS/MS data from non-targeted metabolite profiling studies.&#8221;
<em>Analytical Chemistry</em>, <b>9</b>, 33-43.
 <br /><br />
  Broeckling CD, Afsar FA, Neumann S, Ben-Hur A, Prenni JE (2014).
&#8220;RAMClust: A Novel Feature Clustering Method Enables Spectral-Matching-Based Annotation for Metabolomics Data.&#8221;
<em>Analytical Chemistry</em>, <b>86 (14)</b>, 6812–6817.</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Rick Helmus &lt;<a href='mailto:r.helmus@uva.nl'>r.helmus@uva.nl</a>&gt; and Bas van de Velde (major contributions to spectral binning and
  similarity calculation).</p>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Rick Helmus.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


